### PR TITLE
Various docs changes for CI

### PR DESCRIPTION
**Pull request by woodsp-ibm**
### Summary

See Qiskit/qiskit-nature#1109 for more detail.

Basically does the same three things

- Removes dependency on sphinx-autodoc-typehints
- Adds default thumbnail config as that changed in nbsphinx 0.9
- Updates tox.ini gettext section used for translatable strings due to ipykernel not being installed by default now

### CODE REVIEW

File: tox.ini [hunk 1]
```diff
@@ -58,6 +58,7 @@ deps =
   -r{toxinidir}/requirements.txt
   -r{toxinidir}/requirements-dev.txt
   sphinx-intl
+  jupyter
```
Comment (t-imamichi) [inline tox.ini:61]: Is it necessary? I succeeded to run `tox -egettext` without this line and got the following message.
```
  gettext: OK (97.34=setup[60.66]+cmd[34.39,2.28] seconds)
  congratulations :) (97.44 seconds)
```

Comment (t-imamichi) [reply tox.ini:61]: Thank you for the details. Let's merge them as they are.

**Review by t-imamichi — APPROVED** [review]

**Comment by t-imamichi** [comment]
Is this backport stable?

**Comment by woodsp-ibm** [comment]
> Is this backport stable?

We can do, I have done that in other places. Since the next release may be a little way off yet its probably better to do so.

**Pull request merged**